core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```
@@ -0,0 +1,363 @@
+      case INTEGER:
+        Preconditions.checkArgument(defaultValue.isIntegralNumber() && defaultValue.canConvertToInt(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.intValue();
+      case LONG:
+        Preconditions.checkArgument(defaultValue.isIntegralNumber() && defaultValue.canConvertToLong(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.longValue();
+      case FLOAT:
+        Preconditions.checkArgument(defaultValue.isFloatingPointNumber(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.floatValue();
+      case DOUBLE:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.doubleValue();
+      case DECIMAL:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        BigDecimal retDecimal = defaultValue.decimalValue();
+        Preconditions.checkArgument(
+            retDecimal.scale() == ((Types.DecimalType) type).scale(), "Cannot parse default as a %s value: %s",
+            type,
+            defaultValue);
+        return retDecimal;
+      case STRING:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        return defaultValue.textValue();
+      case UUID:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        try {
+          UUID.fromString(defaultValue.textValue());
+        } catch (IllegalArgumentException e) {
+          throw new IllegalArgumentException(String.format("Cannot parse default as a %s value: %s", type,
+              defaultValue));
```

In Iceberg, you should never suppress an exception that was caught. You can add context, but always wrap the exception.

---

fixed

---

fixed